### PR TITLE
Serverless

### PR DESCRIPTION
**Pull request by djkapner**
added the ability for json or json.gz input/output for any and all of input stack, pointmatches, and output stack. In the case of all, it will be serverless.
In getting this to pass tests on travis, did some changes to try to get rid of memory leaks that came up when running tests over and over again, as during the travis pytests. Now using thread pool instead of multiprocess pool, and some other changes...

### CODE REVIEW

**Comment by djkapner** [comment]
renderapi is another possible place for `jsongz`
i'm importing this and renderapi into em-lens-correction and also using `jsongz` there

File: EMaligner/EMaligner.py [hunk 1]
```diff
@@ -442,7 +428,7 @@ def create_CSR_A(self, resolved):
         tile_ids = np.array([t.tileId for t in resolved.tilespecs])
         fargs = [[pairs[i], i, self.args, tile_ids] for i in range(npairs)]
 
-        with renderapi.client.WithPool(self.args['n_parallel_jobs']) as pool:
+        with stdlib_pool.WithThreadPool(self.args['n_parallel_jobs']) as pool:
```
Comment (RussTorres) [inline EMaligner/EMaligner.py]: if you were to (earlier on) do `renderapi.client.WithPool = stdlib_pool.WithThreadPool`, it should also switch to using threadpools for everything in the renderapi client (e.g. `import_tilespecs_parallel`).  I don't think that has a performance hit and might make a difference on Windows clients.

File: EMaligner/jsongz.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+def dump(obj, filepath, compress=None, encoding='utf-8', *args, **kwargs):
+    if not compress:
+        compress = _check_ext(filepath)
+    filepath = _convert_ext(filepath, compress)
+    if compress:
+        with gzip.GzipFile(filepath, 'w') as f:
+            f.write(json.dumps(obj, *args, **kwargs).encode(encoding))
+    else:
+        with open(filepath, 'w') as f:
```
Comment (RussTorres) [inline EMaligner/jsongz.py]: If encoding does not apply to json files it should be documented.

File: EMaligner/schemas.py [hunk 3]
```diff
@@ -63,8 +65,15 @@ def tolist(self, data):
         if not isinstance(data['name'], list):
             data['name'] = [data['name']]
 
+    @post_load
```
[the remark this answers is not in the document]
Comment (djkapner) [reply EMaligner/schemas.py]: no reason, did it

File: EMaligner/utils.py [hunk 4]
```diff
@@ -121,12 +133,14 @@ def get_resolved_from_z(stack, tform_name, fullsize, order, z):
             pass
     if stack['db_interface'] == 'mongo':
         filt = {'z': float(z)}
-        if dbconnection.count_documents(filt) != 0:
+        if dbconnection[1].count_documents(filt) != 0:
             # this sort ordering is the same as render, I think
-            cursor = dbconnection.find(filt).sort([
+            cursor = dbconnection[1].find(filt).sort([
```
[the remark this answers is not in the document]
Comment (djkapner) [reply EMaligner/utils.py]: good idea, did it

File: EMaligner/utils.py [hunk 5]
```diff
@@ -50,15 +51,17 @@ def make_dbconnection(collection, which='tile', interface=None):
                     '__' + collection['project'] +
                     '__' + collection['name'][0] +
                     '__'+which)
-            dbconnection = client.render[mongo_collection_name]
+            dbconnection = (client, client.render[mongo_collection_name])
```
[the remark this answers is not in the document]
Comment (djkapner) [reply EMaligner/utils.py]: did it

File: EMaligner/EMaligner.py (hunk 1 above)
Comment (djkapner) [reply EMaligner/EMaligner.py]: did this. `import_tilespecs_parallel` still switches into pathos if in python 2 and trying to suppress render output.

File: EMaligner/jsongz.py (hunk 2 above)
Comment (djkapner) [reply EMaligner/jsongz.py]: added docs and made encoding apply to json files also.

**Pull request merged**